demos/digit-prediction/digits_prediction.nlogo

```diff
@@ -0,0 +1,648 @@
+extensions [ py ]
+
+globals [
+  coords-x coords-y color-list digit
+  zero one two three four five six seven eight nine
+]
+
+to setup-py
+  clear-all
+  py:setup py:python
+  (py:run
+    "import numpy as np"
+    "from keras.models import load_model"
+  )
+  ;; load model
+  py:run "model = load_model(\"model.h5\")"
+
+  ;; create coords list, useful to make color_list
+  set coords-x (range max-pxcor (min-pxcor - 1) -1)
+  set coords-y (range max-pycor (min-pycor - 1) -1)
+  set color-list []
+end
+
+to draw
+  ifelse mouse-down?
+  [
+    ask patch mouse-xcor mouse-ycor
+    [
+      ifelse erase?
+      [
+        set pcolor black
+      ]
+      [
+        set pcolor white
+      ]
+    ]
+  ]
+  [
+    predict
```

Comment:
This is calling `predict` whenever the user isn't drawing (that is, `draw` is pressed, but the user isn't actually doing anything). That's a lot of unnecessary calls. You might consider instead putting the `predict` right after the `ask` instead. This will give users real-time predictions, and use significantly less CPU when idling.